File: Resueltos_por_competidor/Gabriella_Castro/IQTest.java
/*
 * Autor: Gabriella Castro
 * Problema: IQ Test
 * Juez online: Codeforces
 * Veredicto: Accepted
 * URL: https://codeforces.com/problemset/problem/25/A
 */

import java.util.Scanner;

public class IQTest {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);

		int n = sc.nextInt();
		int par = 0, impar = 0;
		int idxPar = 0, idxImpar = 0;

		for (int i = 1; i <= n; i++) {
			int x = sc.nextInt();
			if (x % 2 == 0) {
				par++;
				idxPar = i;
			} else {
				impar++;
				idxImpar = i;
			}
		}

		if (par == 1) {
			System.out.println(idxPar);
		} else {
			System.out.println(idxImpar);
		}

	}
}

File: Resueltos_por_competidor/Juan_Martinez/IQTest.cpp
/*
 * Autor: Juan Martinez
 * Problema: IQ test (25A)
 * Juez online: Codeforces
 * Veredicto: Accepted
 * Url: https://codeforces.com/problemset/problem/25/A
 * Difficulty: 1300
 */
#include <bits/stdc++.h>
using namespace std;

int main() {
	int n; cin>>n;
    map<int,int> m;
    int arr[n];
    bool e = 1;
    for(int i = 0; i < n; i++) cin>>arr[i], m[(arr[i]%2)]++;
    if(m[0] > m[1]) e = 0;
    for(int i = 0; i < n; i++){
        if(arr[i]%2 != e) cout<<i+1<<endl;
    }
}
